api/server/httputils/form.go

```diff
@@ -109,3 +115,34 @@ func ArchiveFormValues(r *http.Request, vars map[string]string) (ArchiveOptions,
 	}
 	return ArchiveOptions{name, path}, nil
 }
+
+// ParsePlatform parses the OCI platform from a form
+func ParsePlatform(form url.Values) (*ocispec.Platform, error) {
+	var p ocispec.Platform
+
+	platformJson := form.Get("platform")
+	if platformJson == "" {
+		return nil, nil
+	}
```

Comment:
Slightly wondering if we should move the `form.Get` out of this function, and pass the `platformJson` (lol; probably should be named `platformJSON`).

In that case we could only parse if the `platform` parameter is given, and make the parsing (should it be named "unmarshal?") stricter.

Currently we have a couple of branches that `return nil, nil`, which could be a bit ambiguous to use


When calling it, we would do so conditionally; something like;

```go
var platform *ocispec.Platform
if p := r.Form.Get("platform"); p != "" {
	var err error
	platform, err = httputils.ParsePlatform(p)
	if err != nil {
		return err
	}
}
```